rust/main/submitter/src/chain_tx_adapter/chains/sealevel/adapter.rs

```diff
@@ -316,12 +323,39 @@ impl AdaptsChain for SealevelTxAdapter {
         Ok(success)
     }
 
+    async fn estimate_tx(&self, tx: &mut Transaction) -> Result<(), SubmitterError> {
+        info!(?tx, "estimating transaction");
+        let not_estimated = tx.precursor();
+        let estimated = self.estimate(not_estimated).await?;
```

Comment:
i'm assuming `self.estimate()` also tell us if simulation failed, so we should be able to remove `simulate_tx` from `AdaptsChains`?